lib_nbgl/src/nbgl_layout.c

```diff
@@ -448,13 +448,79 @@ static void spinnerTickerCallback(void)
 
     // get index of obj
     while (i < layout->container->nbChildren) {
-        if (layout->container->children[i]->type == SPINNER) {
-            spinner = (nbgl_spinner_t *) layout->container->children[i];
-            spinner->position++;
-            spinner->position &= 3;  // modulo 4
-            nbgl_objDraw((nbgl_obj_t *) spinner);
-            nbgl_refreshSpecial(BLACK_AND_WHITE_FAST_REFRESH);
-            return;
+        if (layout->container->children[i]->type == CONTAINER) {
+            nbgl_container_t *container = (nbgl_container_t *) layout->container->children[i];
+            if (container->children[0]->type == SPINNER) {
+                spinner = (nbgl_spinner_t *) container->children[0];
+                spinner->position++;
+                spinner->position &= 3;  // modulo 4
+                nbgl_objDraw((nbgl_obj_t *) spinner);
+                nbgl_refreshSpecial(BLACK_AND_WHITE_FAST_REFRESH);
+                return;
+            }
+        }
+        i++;
+    }
+}
+
+// callback for animation ticker
+static void animTickerCallback(void)
+{
+    nbgl_image_t          *image;
+    uint8_t                i = 0;
+    nbgl_layoutInternal_t *layout;
+
+    // gLayout[1] is on top of gLayout[0] so if gLayout[1] is active, it must catch the event
+    if (gLayout[1].nbChildren > 0) {
+        layout = &gLayout[1];
+    }
+    else {
+        layout = &gLayout[0];
+    }
+
+    if (layout->animation == NULL) {
+        return;
+    }
+
+    // get index of image obj
+    while (i < layout->container->nbChildren) {
+        if (layout->container->children[i]->type == CONTAINER) {
+            nbgl_container_t *container = (nbgl_container_t *) layout->container->children[i];
+            if (container->children[1]->type == IMAGE) {
+                image = (nbgl_image_t *) container->children[1];
+                if (layout->animation->parsing == LOOP_PARSING) {
+                    if (layout->iconIdxInAnim == (layout->animation->nbIcons - 1)) {
```

Comment:
This could be a simple ++, with a modulo...

Reply:
Not sure that it would be more readable nor efficient (because not a ^2 static modulo)